app/controllers/reserves_controller.rb

```diff
@@ -49,15 +49,10 @@ def add_item
         if params[:sw] == 'false'
           params[:item] = {}
         elsif params[:sw] == 'true'
-          params[:item] = {}
           ckey = params[:url].strip[/(\d+)$/]
-          url = searchworks_ckey_url("#{ckey}.mobile?covers=false&availability=false")
-          doc = Nokogiri::XML(Faraday.get(url).body)
-          title = doc.xpath("//full_title").text
-          format = doc.xpath("//formats/format").map { |x| x.text }
-          render(text: "alert('This does not appear to be a valid item in SearchWorks'); clean_up_loading();") && return if title.blank?
-          params[:item] = { title: doc.xpath("//full_title").text, ckey: ckey }
-          params[:item].merge!(loan_period: "4 hours", media: "true") if format.include?("Video")
+          item = SearchWorksItem.new(ckey)
+          render(js: "alert('This does not appear to be a valid item in SearchWorks'); clean_up_loading();") && return unless item.valid?
+          params[:item] = item.to_h
```

Comment:
I went down the path of trying to pass just the reference to the item, but unfortunately this is effectively used as a scaffold of the reserve form "row" before it's serialized and stored in the DB.  The view code uses both this data (when a record is first added to a list and rendered) and the data serialized in the database (when a stored list is retrieved and rendered), and it got down into a quite a rabbit hole.  This feels like a small enough surface area and will allow us to add and persist data to the form from SearchWorks w/o thrashing around in controller code.